docs/layout-rendering.md

```
@@ -0,0 +1,207 @@
+- The thought first mounts.
+- The `value` prop changes.
+- The cursor moves (subscribed via `useSelectorEffect(updateSize, selectCursor, isEqual)`).
+- The thought's style changes (subscribed via `useSelectorEffect`).
+- The viewport's `innerWidth` changes.
+- `autofocus`, `crossContextualKey`, `id`, or `isVisible` change.
+
+`offsetHeight` is used on touch (avoids transform-induced fractional pixels); `getBoundingClientRect().height` on desktop.
+
+## Render tree
+
+```
```

markdownlint-cli2 (0.22.1)

[warning] Fenced code blocks should have a language specified

(MD040, fenced-code-language)